lib/ipaddress/ipv4.rb

```diff
@@ -805,6 +805,42 @@ def subnet(subprefix)
       end
     end
 
+    #
+    # Subtract other from the current range
```

Comment:
Can we explain this method in more detail? I have decent knowledge of IPv4 addresses, but I don't really understand the use-case or what this does.

Also, is this limited to ranges? If so, any validation around that? Should that be reflected in naming?

Reply:
I will add some extended explanation. 
This use-case is pretty common if you want to plan network. E.g. you dedicated /18 range for k8s clusters control planes. There are 3 clusters, each of whose occupied /20 range. This method allows to know what ranges are free.

Another use-case is here: https://docs.aws.amazon.com/vpc/latest/userguide/aws-ip-ranges.html#aws-ip-egress-control. It tells "allow all by EC2".

Reply:
> Also, is this limited to ranges? If so, any validation around that? Should that be reflected in naming?

No, it isn't. Individual addresses are essentially ranges of 1 element with /32 netmask.
And current implementation works with individual addresses, I will add test case for that.